api/src/main/java/org/ccci/idm/user/DefaultUserManager.java

```diff
@@ -307,4 +337,28 @@ protected void validateUser(final User user) throws UserException {
             throw new InvalidEmailUserException("Invalid email specified for user");
         }
     }
+
```

Comment:
Do you prefer defining the interface and abstract class herein? I think I would've created them in separate files, interface and class definition unto themselves. I would do so so as not to 'clutter' this class definition (though you might not see it as clutter). Is there some guiding principle wherein you find it more meaningful to include definitions herein?